src/storage/src/mut_records.rs

```diff
@@ -195,6 +163,32 @@ impl FileRecords for MutFileRecords {
     }
 }
 
+impl MutFileRecordsFlushPolicy {
+    /// Evaluates the flush policy and returns true
+    /// if the policy determines the need to flush
+    fn should_flush(&mut self) -> bool {
+        use MutFileRecordsFlushPolicy::*;
+
+        match self {
+            NoFlush => false,
+
+            EveryWrite => true,
+
+            CountWrites {
+                n_writes: n_max,
+                write_tracking: wcount,
```

Comment:
write_tracking seems to be duplicate of `write_count`.  Do we need both?

Reply:
I made write_count an absolute count tied to mut_records for metrics, write tracking is tied to the flush policy but is reset to zero after a flush. I could have unified them and triggered the flush policy on a (write_count % n_writes == 0), but thought the divide would be more expensive than two counters and two increments.

Reply:
Sounds good